freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/FreeIpaClientFactory.java

```diff
@@ -87,6 +88,7 @@ public FreeIpaClient getFreeIpaClientForStackId(Long stackId) throws FreeIpaClie
 
     public FreeIpaClient getFreeIpaClientByAccountAndEnvironment(String environmentCrn, String accountId) throws FreeIpaClientException {
         Stack stack = stackService.getByEnvironmentCrnAndAccountId(environmentCrn, accountId);
+        MDCBuilder.buildMdcContext(stack);
```

Comment:
would it make sense to move this to the private `getFreeIpaClient` method? Then it would work for all the various public methods of this class.

Reply:
I have added MDC building where the context was insufficient. I started from the endpoints and followed the call chain to check if the context is set at the earliest point where `Stack` is available. Most probably the calls to the `FreeIpaClientFactory` come from flows, which have the context already set (at least I hope so, haven't checked all of them). But it's possible I missed some places so I'm open to suggestions :)

Reply:
it makes sense to set the MDC at the earliest possible moment. The `FreeIpaClientFactory` should be able to rely on callers of `getFreeIpaClientForStack` to have already set the MDC.